src/rpc/blockchain.cpp

```diff
@@ -1989,18 +1988,15 @@ class CoinsViewScanReserver
 
     bool reserve() {
         CHECK_NONFATAL(!m_could_reserve);
-        std::lock_guard<std::mutex> lock(g_utxosetscan);
-        if (g_scan_in_progress) {
+        if (g_scan_in_progress.exchange(true)) {
             return false;
         }
-        g_scan_in_progress = true;
         m_could_reserve = true;
         return true;
     }
```

Comment:
c3cf2f55013c4ea1c1ef4a878fc7ff8e92f2c42d
It seems this changes behavior as `m_could_reserve = true;` is also guarded by mutex currently.

Reply:
`m_could_reserve` isn't a global, and instances of `CoinsViewScanReserver` aren't shared across threads either.

Reply:
Oh, I see that all instances of `CoinsViewScanReserver` are local in the `scantxoutset()` function.